ideascube/serveradmin/wifi.py

```diff
@@ -218,3 +218,67 @@ def get_wifi_device():
             _("Found more than one Wi-Fi device, this is not supported"))
 
     return devices[0]
+
+
+class WPAConfig(object):
+
+    options = {
+         'wpa': '2',
+         'wpa_passphrase': '',
+         'wpa_key_mgmt': 'WPA-PSK',
+         'wpa_pairwise': 'TKIP',
+         'rsn_pairwise': 'CCMP',
+    }
+
+    def __init__(self, configfile='/etc/hostapd/hostapd.conf'):
+        self.is_enable = False
+        # read data from config
+        with open(configfile) as fd:
+            self.data = fd.readlines()
+        # update option with current passphrase and check if WPA is enable
+        for line in self.data:
+            if '=' in line and 'wpa_passphrase' in line:
+                 self.is_enable = not line.startswith('#')
+                 line = line.strip('# ')
+                 passphrase = line.split('=', 1)[1].strip()
+                 self.options['wpa_passphrase'] = passphrase
+        self.new_data = self.data[:]
+        self.configfile = configfile
+
+    def save(self):
+        """save data to configfile"""
+        with open(self.configfile, 'w') as fd:
+            fd.write('\n'.join(self.new_data))
```

Comment:
What we did is to create a group with the ideascube user and give write permission to that group on that file. It will come in another PR about the Ansible recipe.